client/cook/client.py

```diff
@@ -0,0 +1,201 @@
+
+import time
+import uuid
+import functools
+
+from bravado_core.formatter import SwaggerFormat
+from bravado_core.exception import SwaggerValidationError
+
+from bravado.client import SwaggerClient
+from bravado.requests_client import RequestsClient
+
+class hashabledict(dict):
+    def __hash__(self):
+        return hash(frozenset(self))
+
+def basic_auth_http_client(*args):
+    http_client = RequestsClient()
+    http_client.set_basic_auth(*args)
+
+    return http_client
+
+def key_auth_http_client(*args):
+    http_client = RequestClient()
+    http_client.set_api_key(*args)
+
+    return http_client
+
+def uuid_to_python(v):
+    return uuid.UUID(v)
+
+def uuid_to_wire(v):
+    return str(v)
+
+def validate_uuid(v):
+    try:
+        return str(uuid.UUID(v)) == v
+    except:
+        return False
+
+UUID_SWAGGER_FORMAT=SwaggerFormat(
+    format='uuid',
+    description='format and validate uuid values',
+    to_python=uuid_to_python,
+    to_wire=uuid_to_wire,
+    validate=validate_uuid
+)
+
+class CookClient:
+    def __init__(self, **kwargs):
+        if kwargs.get('url'):
+            self.client = SwaggerClient.from_url(
+                kwargs.get('url'),
+                http_client=kwargs.get('http_client'),
+                config={'also_return_response': True}
+            )
+        elif kwargs.get('file'):
+            self.client = SwaggerClient.from_spec(
+                load_file(kwargs.get('file')),
+                http_client=kwargs.get('http_client'),
+                config={'also_return_response': True}
+            )
+        elif kwargs.get('spec'):
+            self.client = SwaggerClient.from_spec(
+                kwargs.get('spec'),
+                http_client=kwargs.get('http_client'),
+                config={'also_return_response': True}
+            )
+
+        self.client.swagger_spec.register_format(UUID_SWAGGER_FORMAT)
+
+    def schedule_job(self, job):
+        return self.schedule_jobs([job])[0]
+
+    def schedule_jobs(self, jobs):
+        _, result = self.client.rawscheduler.post_rawscheduler(
+            job={"jobs": jobs}
+        ).result()
+
+        return result.json()
+
+    def delete_job(self, uuid):
+        return self.delete_jobs([uuid])
+
+    def delete_jobs(self, uuids):
+        result, _ = self.client.rawscheduler.delete_rawscheduler(
+            job=uuids
+        ).result()
+
+        return result is None
+
+    def poll_job(self, uuid):
+        return self.poll_jobs([uuid])[0]
+
+    def poll_jobs(self, uuids):
+        _, result = self.client.rawscheduler.get_rawscheduler(
+            job=uuids
+        ).result()
+
+        return result.json()
+
+    def await_job(self, uuid, t=30):
```

Comment:
Add a parameter for how frequently to poll. For many jobs, 1 second is probably way too frequent, 30 seconds is probably a good default.